### PR TITLE
Introduce muzzle-specific interface to InstrumentationModule

### PR DESCRIPTION
**Pull request by iNikem**
It will contain all methods that muzzle automatically generates.

### CODE REVIEW

File: muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java [hunk 1]
```diff
@@ -45,6 +45,11 @@ protected MethodHandle computeValue(Class<?> type) {
    */
   @SuppressWarnings("unchecked")
   static Map<String, ClassRef> getFor(InstrumentationModule instrumentationModule) {
+    if (instrumentationModule instanceof InstrumentationModuleMuzzle) {
+      return ((InstrumentationModuleMuzzle) instrumentationModule).getMuzzleReferences();
+    }
+
+    //Older classes created and compiled outside of this repo may not yet have the interface above.
```
Comment (iNikem) [inline muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java:52]: Not sure if we should be backward-compatible here...

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/InstrumentationModuleInstaller.java [hunk 2]
```diff
@@ -120,20 +121,23 @@ AgentBuilder install(
...
       InstrumentationModule instrumentationModule) {
-    InstrumentationContextBuilderImpl builder = new InstrumentationContextBuilderImpl();
-    instrumentationModule.registerMuzzleContextStoreClasses(builder);
-    ContextStoreMappings mappings = builder.build();
-    if (!mappings.isEmpty()) {
-      return FieldBackedProviderFactory.get(instrumentationModule.getClass(), mappings);
-    } else {
-      return NoopContextProvider.INSTANCE;
+
+    if (instrumentationModule instanceof InstrumentationModuleMuzzle) {
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/InstrumentationModuleInstaller.java:125]: may be worth a `debug` log in an `else`?

File: muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java [hunk 3]
```diff
@@ -45,6 +45,11 @@ protected MethodHandle computeValue(Class<?> type) {
    */
   @SuppressWarnings("unchecked")
   static Map<String, ClassRef> getFor(InstrumentationModule instrumentationModule) {
+    if (instrumentationModule instanceof InstrumentationModuleMuzzle) {
+      return ((InstrumentationModuleMuzzle) instrumentationModule).getMuzzleReferences();
+    }
+
+    // Older classes created and compiled outside of this repo may not yet have the interface above.
```
Comment (trask) [inline muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java:52]: how about log at `warn` level below in `methodHandle != null`, saying that this path is deprecated and will be removed in the near future?

**Review by trask — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]
Nice 💯

**Pull request merged**